### PR TITLE
[Merge pending] add search ratio for limit forwarding search k size for agent & add disable_async_rerank option to use primitive sort rerank

### PR DESCRIPTION
**Pull request by kpango**
### Description:
This PR includes 2 feature.
1. Efficient Search-K forwarding for Agent using Query ratio paramerter.
         - Breaking: user should set 1 (100%) as default search query.
         - **This PR requires minor update due to the breaking changes to proto payload.**
2. Add non-async merging sort for low Top-K with low pod replica users which may faster to use lock and append and sort.

### Versions:

- Go Version: 1.20.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 2.0.9

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

**Comment by vankichi** [comment]
Could you please tell me the background and the benefits at the PR description?

File: apis/proto/v1/payload/payload.proto [hunk 1]
```diff
@@ -79,18 +79,22 @@ message Search {
...
+    // Disables asynchronous reranking
+    bool disable_async_rerank = 8;
     // Ingress filter configurations.
-    Filter.Config ingress_filters = 6;
+    Filter.Config ingress_filters = 11;
     // Egress filter configurations.
-    Filter.Config egress_filters = 7;
-    // Minimum number of result to be returned.
-    uint32 min_num = 8 [ (validate.rules).uint32.gte = 0 ];
+    Filter.Config egress_filters = 12;
```
Comment (vankichi) [inline apis/proto/v1/payload/payload.proto:97]: If I remember correctly, you told me that proto parameters should be added later, but what is the reason for rewriting this time?

File: pkg/gateway/lb/handler/grpc/handler.go [hunk 2]
```diff
@@ -475,6 +497,22 @@ func (s *server) SearchByID(ctx context.Context, req *payload.Search_IDRequest)
 	return res, nil
 }
 
+func (s *server) calculateNum(ctx context.Context, num uint32, ratio float32) (n uint32) {
+	min := float64(s.replica) / float64(len(s.gateway.Addrs(ctx)))
+	if ratio < 0.0 {
```
Comment (vankichi) [inline pkg/gateway/lb/handler/grpc/handler.go:502]: It may this ? :thinking: 
```suggestion
	if ratio > 0.0 {
```

Comment (kpango) [reply pkg/gateway/lb/handler/grpc/handler.go:502]: why?

Comment (vankichi) [reply pkg/gateway/lb/handler/grpc/handler.go:502]: ah, sorry, it's my misunderstanding. :sweat:

**Comment by kpango** [comment]
@vankichi 
> Could you please tell me the background and the benefits at the PR description?

Due to the changes in the Search specification, the usage of the User's Client library will change, so a " Minor" equivalent release is required (Client backward incompatibility), and the Proto Index has been reorganized in this release.

**Pull request closed without merging**